unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Textures/ComponentTextureAttachment.cs

```diff
@@ -0,0 +1,201 @@
+﻿using System;
+using DCL.Models;
+using UnityEngine;
+
+namespace DCL.Components
+{
+    public class ComponentTextureAttachment : ITextureAttachment
+    {
+        public ISharedComponent component => disposableComponent;
+
+        private BaseDisposable disposableComponent;
+
+        private Action<ITextureAttachment> OnUpdateEvent;
+
+        // As the SubscribeToEntityUpdates() method is not easy on the CPU, we only call it if the OnUpdateEvent
+        // is actually used. this means that DCLVideoTexture will still work without incurring overhead to
+        // DCLTexture components.
+        public event Action<ITextureAttachment> OnUpdate
+        {
+            add
+            {
+                if ( OnUpdateEvent == null || OnUpdateEvent.GetInvocationList().Length == 0 )
```

Comment:
What happens if the disposableComponent gets attached to a new entity but `OnUpdate +=` is not called again? We will miss an update from that entity?